### PR TITLE
[Documentation][CatalogPromotion] Add cookbook about creating a custom action + minor improvements

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | partially fixes #https://github.com/Sylius/Sylius/pull/13325#discussion_r756750238
| License         | MIT

This PR contains a minor refactor of parameters with types of actions/scopes passed to validators.

### CODE REVIEW

File: docs/cookbook/promotions/custom-catalog-promotion-action.rst [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+                    return;
+                }
+            }
+        }
+    }
+
+Alright we have a working basic validation, and our new type of action exists and can be created, and edited
+programmatically or by API. Let's now prepare a UI part of this new feature.
+
+Prepare a configuration form type for a new action
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-action.rst:183]:
```suggestion
Prepare a configuration form type for the new action
```

File: docs/cookbook/promotions/custom-catalog-promotion-action.rst [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+--------------------------------------------------
+
+To be able to configure a catalog promotion with your new action you will need a form type for the admin panel.
+And with current implementation, as our action is channel based, you need to create 2 form types as below:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-action.rst:187]:
```suggestion
To be able to configure a catalog promotion with your new action you will need a form type for the admin panel.
And with the current implementation, as our action is channel-based, you need to create 2 form types as below:
```

File: docs/cookbook/promotions/custom-catalog-promotion-action.rst [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+        {% for channelCode, channelConfiguration in action.configuration %}
+            <tr>
+                <td class="five wide"><strong class="gray text">{{ channelCode }}</strong></td>
+                <td>{{ money.format(channelConfiguration.price, currencies[channelCode]) }}</td>
+            </tr>
+        {% endfor %}
+        </tbody>
+    </table>
+
+That's all. You will now should be able to choose the new action while creating or editing a catalog promotion.
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-action.rst:310]:
```suggestion
That's all. You will now be able to choose the new action while creating or editing a catalog promotion.
```

File: docs/cookbook/promotions/custom-catalog-promotion-scope.rst [hunk 4]
```diff
@@ -30,66 +30,20 @@ The new Scope needs to be declared somewhere, it would be nice to extend the cur
         public const TYPE_BY_PHRASE = 'by_phrase';
     }
 
-Now lets declare the basic validator service, with added additional Scope. This first validation is necessary to check
-if the Scope is added to Catalog Promotion and also if the mandatory key fields exist. We will also declare a more
-atomic validator for our Scope:
+Now let's declare the parameter with scope types, with added our additional custom scope as the last one:
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-scope.rst:33]:
```suggestion
Now let's declare the parameter with scope types, with our additional custom scope added as the last one:
```

File: docs/cookbook/promotions/custom-catalog-promotion-scope.rst [hunk 5]
```diff
@@ -147,7 +101,50 @@ And the code for the provider itself:
     or extend the code from this cookbook to e.g. consume key ``localeCode`` from configuration.
 
 Now the Catalog Promotion should work with your new Scope for programmatically and API created resource.
-Lets now prepare a validation for UI part by using form types.
+
+Prepare a custom validator for a new scope
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-scope.rst:105]:
```suggestion
Prepare a custom validator for the new scope
```

File: docs/cookbook/promotions/custom-catalog-promotion-scope.rst [hunk 6]
```diff
@@ -147,7 +101,50 @@ And the code for the provider itself:
...
+            Assert::isInstanceOf($constraint, CatalogPromotionScope::class);
+
+            if (!array_key_exists('phrase', $configuration) || empty($configuration['phrase'])) {
+                $context->buildViolation('There is no phrase provided')->atPath('configuration.phrase')->addViolation();
+            }
+        }
+    }
+
+Alright we have a working basic validation, and our new type of scope exists and can be created, and edited
+programmatically or by API. Let's now prepare a UI part of this new feature.
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-scope.rst:147]:
```suggestion
Alright, we have a working basic validation, and our new type of scope exists, can be created, and edited
programmatically or by API. Let's now prepare the UI part of this new feature.
```

File: docs/cookbook/promotions/custom-catalog-promotion-scope.rst [hunk 7]
```diff
@@ -218,7 +215,30 @@ In our case - we have a template ``by_phrase.html.twig`` which is first before o
...
+        </tr>
+        <tr>
+            <td class="five wide"><strong class="gray text">Phrase</strong></td>
+            <td>{{ scope.configuration.phrase }}</td>
+        </tr>
+        </tbody>
+    </table>
+
+
+That's all. You will now should be able to choose the new scope while creating or editing a catalog promotion.
```
Comment (CoderMaggie) [inline docs/cookbook/promotions/custom-catalog-promotion-scope.rst:241]:
```suggestion
That's all. You will now be able to choose the new scope while creating or editing a catalog promotion.
```

**Review by AdamKasp — APPROVED** [review]

**Comment by AdamKasp** [comment]
Thanks, Grzegorz! :1st_place_medal:

**Pull request merged**